File: ios/MRT/Core/Models/ChatMessage.swift
import Foundation

struct ChatMessage: Identifiable, Equatable {
    enum Role: String, Equatable {
        case assistant
        case system
    }

    let id: UUID
    let sessionID: String?
    var content: String
    var isComplete: Bool
    let role: Role

    init(
        id: UUID = UUID(),
        sessionID: String?,
        content: String,
        isComplete: Bool,
        role: Role
    ) {
        self.id = id
        self.sessionID = sessionID
        self.content = content
        self.isComplete = isComplete
        self.role = role
    }
}

File: android/app/src/main/java/com/mrt/app/core/models/ChatMessage.kt
package com.mrt.app.core.models

import java.util.UUID

data class ChatMessage(
    val id: UUID = UUID.randomUUID(),
    val sessionId: String?,
    val content: String,
    val isComplete: Boolean,
    val role: Role,
) {
    enum class Role {
        ASSISTANT,
        SYSTEM,
    }
}
